x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/string/UnaryStringFunction.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.sql.expression.function.scalar.string;
+
+import org.elasticsearch.xpack.sql.expression.Expression;
+import org.elasticsearch.xpack.sql.expression.FieldAttribute;
+import org.elasticsearch.xpack.sql.expression.function.scalar.UnaryScalarFunction;
+import org.elasticsearch.xpack.sql.expression.function.scalar.processor.definition.ProcessorDefinition;
+import org.elasticsearch.xpack.sql.expression.function.scalar.processor.definition.ProcessorDefinitions;
+import org.elasticsearch.xpack.sql.expression.function.scalar.processor.definition.UnaryProcessorDefinition;
+import org.elasticsearch.xpack.sql.expression.function.scalar.script.ScriptTemplate;
+import org.elasticsearch.xpack.sql.expression.function.scalar.string.StringProcessor.StringOperation;
+import org.elasticsearch.xpack.sql.tree.Location;
+import org.elasticsearch.xpack.sql.util.StringUtils;
+
+import java.util.Locale;
+import java.util.Objects;
+
+import static java.lang.String.format;
+import static org.elasticsearch.xpack.sql.expression.function.scalar.script.ParamsBuilder.paramsBuilder;
+
+public abstract class UnaryStringFunction extends UnaryScalarFunction {
+
+    protected UnaryStringFunction(Location location, Expression field) {
+        super(location, field);
+    }
+
+    @Override
+    public boolean foldable() {
+        return field().foldable();
+    }
+
+    @Override
+    public Object fold() {
+        return operation().apply(field().fold());
+    }
+
+    @Override
+    protected TypeResolution resolveType() {
```

Comment:
When I've made changes to there I've been *trying* to make sure methods in these were either `abstract` or `final`. That sort of forces things to be the same as one another and not "almost the same". Like `Char` doesn't feel like a real `UnaryStringFunction` to me because it overrides this method.

Reply:
@nik9000 agree with you that it doesn't feel like a true unary string function, judging by the type of the parameter. And it would be an awkward move to override the method checking the parameter type for this reason. But in SQL, `CHAR` and `SPACE` (the only two functions that do not receive at least one string as parameter) are considered to be string functions probably because they create text in one form or another, irrespective of the parameters count and type. Do you think it's worth taking these two out of the UnaryStringFunction base class and handle them differently?

Reply:
Personally, yeah, I'd stop extending that class. Or at least see what that looks like.

Reply:
@nik9000 I've made the changes and introduced a new `UnaryStringIntFunction` class to handle the Char and Space functions. Let me know what you think.